### PR TITLE
Replace print statements with rich in pyodide-build

### LINKED ISSUE #3430: On first call, `pyodide config get emscripten_version` returns `Downloading xbuild environment Installing xbuild environment 3.1.27` instead of `3.1.27`
## 🐛 Bug

In [the docs for out-of-tree builds](https://pyodide.org/en/stable/development/building-and-testing-packages.html#building-and-testing-packages-out-of-tree) it gives this code snippet:

```bash
pip install pyodide-build

git clone https://github.com/emscripten-core/emsdk.git
cd emsdk

PYODIDE_EMSCRIPTEN_VERSION=$(pyodide config get emscripten_version)
./emsdk install ${PYODIDE_EMSCRIPTEN_VERSION}
./emsdk activate ${PYODIDE_EMSCRIPTEN_VERSION}
source emsdk_env.sh
```
But this doesn't work because on the first call, `pyodide config get emscripten_version` outputs this:
```
Downloading xbuild environment
Installing xbuild environment
3.1.27
```
On subsequent calls it returns `3.1.27`.

### To Reproduce

See above.

### Expected behavior

Calls to `pyodide config get emscripten_version` should only ever output the version string such that this command can be reliably used in build automation.

### Environment

- Pyodide Version: Pyodide CLI Version: 0.2.2

### Additional context

As a workaround for build scripts, `pyodide config get emscripten_version` can be called once before actually using it.

### PR DESCRIPTION
**Pull request by ryanking13**
### Description

Close: #2074
Close: #2076

### Checklists

- [x] Use logger instead of naive print
- [x] Apply changes from #2076
- [x] mkpkg
- [x] buildall
- [x] common
- [x] cli
- [x] buildpkg
- [x] xbuildenv
- [x] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry

pypabuild: no modification to make it compatible with pypa/build output
pywasmcross: no modification to not introduce additional dependencies to pywasmcross

### CODE REVIEW

**Comment by ryanking13** [comment]
I think this is ready to be reviewed.

- Thanks to the excellent work by @hoodmane (#2076), the package build status bar in CI now [looks good](https://app.circleci.com/pipelines/github/ryanking13/pyodide/1443/workflows/062a707c-aca5-4e03-a7fd-11daac14c214/jobs/26584?invite=true#step-104-9).
- I changed warning / error level logs to be printed to stderr, and debug / info level logs to be printed to stdout. To be honest, I don't know much about the logging convention, so please let me know if I am not following any common conventions (colors, levels, etc).

File: pyodide-build/pyodide_build/buildall.py [hunk 1]
```diff
@@ -158,15 +164,95 @@ def build(self, args: Any) -> None:
             )
 
         if p.returncode != 0:
-            print(f"Error building {self.name}. Printing build logs.")
+            logger.error(f"Error building {self.name}. Printing build logs.")
 
             with open(self.pkgdir / "build.log") as f:
-                shutil.copyfileobj(f, sys.stdout)
+                shutil.copyfileobj(f, sys.stderr)
```
Comment (hoodmane) [inline pyodide-build/pyodide_build/buildall.py:170]: Maybe for consistency we could write this to `logger.error`?

File: pyodide-build/pyodide_build/buildall.py [hunk 2]
```diff
@@ -158,15 +164,95 @@ def build(self, args: Any) -> None:
...
+        self.task = self.progress.add_task("Building packages...", total=num_packages)
+        self.packages: list[PackageStatus] = []
+        self.reset_grid()
+
+    def reset_grid(self):
+        """Empty out the rendered grids."""
+        self.top_grid = Table.grid()
+
+        for package in self.packages:
+            if not package.finished:
```
Comment (hoodmane) [inline pyodide-build/pyodide_build/buildall.py:227]: Since we call `remove_package` immediately after `finish`, this conditional probably isn't needed?

**Review by hoodmane — APPROVED** [review]
Thanks @ryanking13!

**Review by rth — APPROVED** [review]
Thanks for doing this, LGTM as well. Does this supersede https://github.com/pyodide/pyodide/pull/2076 then?

**Comment by hoodmane** [comment]
Sure looks like.

File: pyodide-build/pyodide_build/install_xbuildenv.py [hunk 3]
```diff
@@ -27,7 +28,7 @@ def download_xbuildenv(version: str, xbuildenv_path: Path) -> None:
     from shutil import rmtree, unpack_archive
     from tempfile import NamedTemporaryFile
 
-    print("Downloading xbuild environment")
+    logger.stderr("Downloading xbuild environment")
```
[the remark this answers is not in the document]
Comment (ryanking13) [reply pyodide-build/pyodide_build/install_xbuildenv.py:31]: Good point. I wanted to fix #3430 by printing this message to stderr, but I guess adding a `verbose` parameter instead would be better and consistent.

**Comment by ryanking13** [comment]
Thanks for the reviews!

**Pull request merged**